### PR TITLE
Add support for CORS in http requests

### PR DESCRIPTION
**Pull request by Kilowhisky**
Please do not open a pull request without first filing an issue and/or discussing the feature directly with the project maintainer.

### Please ensure you adhere to every item in this list

- [ ] This PR was pre-approved by the project maintainer
- [x] I have self-reviewed the code
- [x] I have added all necessary tests

### Describe your changes

Adds support for OPTIONS HTTP request and returns necessary headers to allow CORS to function properly

### Issue number and link

https://github.com/tidwall/tile38/issues/737

I'm not quite sure how testing should be performed here as I don't see any testing of the http interface right now.. maybe i'm missing something. Could use a pointer on where and how you'd like me to write tests. 

Tests can be done with standard HTTP requests.

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -1444,6 +1445,21 @@ func readNextHTTPCommand(packet []byte, argsIn [][]byte, msg *Message, wr io.Wri
...
+			}
+			corshead := "HTTP/1.1 204 No Content\r\n"+
+				"Connection: close\r\n"+
+				"Access-Control-Allow-Origin: *\r\n"+
+				"Access-Control-Allow-Methods: POST, GET, OPTIONS\r\n\r\n"
+
+			if _, err = wr.Write([]byte(corshead)); err != nil {
+				return false, err
+			}
+			return false, nil
```
Comment (Kilowhisky) [inline internal/server/server.go:1461]: This seems to work if I return false or true. Not sure which is correct.

**Comment by Kilowhisky** [comment]
Tests added

**Comment by Kilowhisky** [comment]
Verified against my test project in chrome. Request works successfully. 

Screenshot 2024-05-06 at 10 12 56 PM
ub.com/tidwall/tile38/assets/7075666/d56f15c8-3275-4f32-9c69-1e356e68bad5">
Screenshot 2024-05-06 at 10 14 50 PM

**Comment by tidwall** [comment]
I'm cool with this. 🎸

**Pull request merged**